### PR TITLE
OpenAPI v3 CRD schema generation

### PR DESCRIPTION
**Pull request by kron4eg**
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
ref #7398

**Special notes for your reviewer**:

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
OpenAPI v3 CRD schema generation
```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -67,3 +67,10 @@ rm /tmp/headerfile
 
 echodate "Generating reconciling functions"
 go generate pkg/resources/reconciling/ensure.go
+
+echodate "Generating openAPI v3 CRDs"
+go run sigs.k8s.io/controller-tools/cmd/controller-gen \
+  crd \
+  object:headerFile=./hack/boilerplate/ce/boilerplate.go.txt \
```
Comment (xrstf) [inline hack/update-codegen.sh:71]: Where is the headerFile actually used? I don't see it in the generated YAML files.

File: pkg/apis/kubermatic/v1/datacenter.go [hunk 2]
```diff
@@ -23,29 +23,27 @@ import (
...
-	// More info: https://git.k8s.io/community/contributors/devel/api-conventions.md
-	Items []Seed `json:"items" protobuf:"bytes,2,rep,name=items"`
+	Items []Seed `json:"items"`
 }
 
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +kubebuilder:object:generate=true
+// +kubebuilder:object:root=true
 
 // Seed is the type representing a SeedDatacenter
```
Comment (xrstf) [inline pkg/apis/kubermatic/v1/datacenter.go]: `s/Datacenter//`. Otherwise it's super confusing.

File: hack/update-codegen.sh (hunk 1 above)
Comment (kron4eg) [reply hack/update-codegen.sh:71]: in `pkg/apis/kubermatic/v1/zz_generated.deepcopy.go`, without specifying `headerFile=`, that file is loosing its header

**Review by xrstf — CHANGES_REQUESTED** [review]
Minor things

File: pkg/apis/kubermatic/v1/datacenter.go (hunk 2 above)
Comment (kron4eg) [reply pkg/apis/kubermatic/v1/datacenter.go]: Done

**Review by xrstf — APPROVED** [review]
/approve

:tada: So looking forward to this new world of schemas, specs, decks and ducks. Maybe not ducks.

**Pull request merged**